### ISSUE TITLE
Mac version gives error at launch

### ISSUE THREAD

**Issue by Simboiss** [ValveSoftware/Source-1-Games#5625]
After the April 18, 2024 update, the following error occurs on Mac:

-----
An error occurred while launching this game : missing executable
/Users/sim/Library/Application Support/Steam/steamapps/common/Team Fortress
2/hl2_osx
-----

OS version : macOS Mojave, 10.14.6
Steam version : 1709846872

I checked the integrity of installed files, and Steam didn't detect any error. If I check manually in the source folder for Team Fortress 2, like the error says, the executable is, indeed, not there.

I deliberately kept this computer OS not updated because if I do, I can't run Team Fortress 2 anymore, because of the problem of compatibility with 32-bits apps. MacOS Mojave is the last macOS version that can run 32-bit apps.

My main machine is a Mac Studio M1 Max, running macOS Monterey 12.7.3. If I try to launch TF2 from this machine, I get the same error as above. Also, Steam still displays the red banner with the warning:

-----
Your current macOS version is unable to run 32-bit games. This game may not run.
-----

**Comment by AwesomeCoder412412** [comment]
To my knowledge, there simply is no Mac executable anymore. Mac support is dead and buried.

**Comment by pnelego** [comment]
> To my knowledge, there simply is no Mac executable anymore. Mac support is dead and buried.

why do they advertise it then?

**Comment by AwesomeCoder412412** [comment]
> > To my knowledge, there simply is no Mac executable anymore. Mac support is dead and buried.
> 
> 
> 
> why do they advertise it then?

That's fair, valve is just lazy and doesn't want to bother. A 32-bit mac version could possibly still be being provided but I'm doubtful. I wouldn't expect it to work.

**Comment by ethanholt1** [comment]
I would expect a macOS binary in the coming days, if there is one at all.
Valve has retained the Mac compatibility label for the game on the Steam page, so take that with what you will.

**Comment by seanmcgeehan** [comment]
I'll lend my voice to this. Its quite sad to me that a 64 bit mac version wasnt shipped. I think it would easily be a 5% player boost which would make the development economically worth it.

**Comment by Tf2onmac** [comment]
THIS IS SUCK

**Comment by ethanholt1** [comment]
> THIS IS SUCK

Did you make an account just to post that? I respect the grind.
But yeah, I would really like a 64 bit Mac binary, even if it's not signed, or even if it's the 32 bit version still, please bring back compatibility

**Comment by Niterux** [comment]
![image](https://github.com/ValveSoftware/Source-1-Games/assets/110568869/24005c9b-eca4-4707-a334-7ce670d1a7fb)

**Comment by msikma** [comment]
> To my knowledge, there simply is no Mac executable anymore. Mac support is dead and buried.

I've not seen an official statement or indication to this effect, so I don't think we can draw this conclusion so confidently.

Either a 64 bit version will be provided in a future update, or Valve should the Mac compatibility label from the game entirely.

With that being said, people have been able to compile older versions of the Source engine on modern Mac, so it is at least possible and probably not a huge step from the work they've already done to provide Windows/Linux support. It'd be a shame if they don't add Mac support at this point, since it's probably going to be the last time ever that significant work will be done on this game.

**Comment by ChampionCynthia** [comment]
> > To my knowledge, there simply is no Mac executable anymore. Mac support is dead and buried.
> 
> I've not seen an official statement or indication to this effect, so I don't think we can draw this conclusion so confidently.
> 
> Either a 64 bit version will be provided in a future update, or Valve should the Mac compatibility label from the game entirely. I'd give it a week or two.
> 
> With that being said, people have been able to compile older versions of the Source engine on modern Mac, so it is at least possible and probably not a huge step from the work they've already done to provide Windows/Linux support. It'd be a shame if they don't add Mac support at this point, since it's probably going to be the last time ever that significant work will be done on this game.

The lack of an official statement is certainly weird. However, on a technical level there is currently nothing being done for the Mac. Since macOS 10.14 is technically out-of-support since February 15 2024, I doubt anything will be done to fix the 32-bit Mac version. The x64 port is not actually the work of Valve themselves and the ones in charge of that port have mentioned that they aren't working on the Mac build (unless we've been trolled). So unless Valve is actually doing something regarding the Mac build on their own, there's probably nothing going on.

They might end up saying to use Crossover and leave it at that.

Either way, I agree that they should get rid of the Mac compatibility label but considering they didn't remove it for HL1 when they broke it with the 25th Anniversary Update (although there's a Beta you can go to for Mac), it's unlikely to happen.

**Comment by ethanholt1** [comment]
> The lack of an official statement is certainly weird. However, on a technical level there is currently nothing being done for the Mac. Since Steam on macOS 10.14 is technically out-of-support since February 15 2024, I doubt anything will be done to fix the 32-bit Mac version. The x64 port is not actually the work of Valve themselves and the ones in charge of that port have mentioned that they aren't working on the Mac build (unless we've been trolled). So unless Valve is actually doing something regarding the Mac build on their own, there's probably nothing going on.
> 
> They might end up saying to use Crossover and leave it at that.
> 
> Either way, I agree that they should get rid of the Mac compatibility label but considering they didn't remove it for HL1 when they broke it with the 25th Anniversary Update (although there's a Beta you can go to for Mac), it's unlikely to happen.
> 
> TL;DR Synchronize your death watches.

Well, with Counter Strike, there’s the pre 20th anniversary beta that you can use to run the game.
No such thing with TF2. I don’t expect a binary, but at the very least I do expect a beta branch for those of us still on 32 bit machines/people on Macs.
But are servers still compatible with 32 bit clients? It’s a lot to think about.

**Comment by Simboiss** [comment]
I will keep and eye on this thread. For the record, I'm still up to offer my technical help for anything related to the Mac version of Team Fortress 2, for free. I know it's probably not the adequate channel to express this, but I don't know where or who I am supposed to contact, so if any admin or Valve employees sees this, please forward to the appropriate person.

**Comment by Wadmodder** [comment]
Valve will also likely remove all the Mac files for their other games that ran on GoldSrc, Source 1 & Source 2 from their Steam depot download servers very soon if they haven't already. Valve will likely drop macOS support from GoldSrc, Source 1 & Source 2 likely later this decade, since they are no longer interested with the platform because of Apple for altering the ecosystem, such as only supporting the Metal API and the Silicon Mac systems, since Apple wants both Vulkan and MoltenVK on macOS to fail NOT succeed. Besides, macOS 15 is just right around the corner of being announced, meaning that Intel Mac hardware support will likely be killed off in macOS 15 this year, maybe even with macOS versions 16, 17 & 18 when they get announced at WWDC 2025, 2026 & 2027 respectively.

In otherwords, the only income that Valve will likely have with the macOS platform in the future is only their Steam client for third-party games that aren't powered by GoldSrc, Source 1 and Source 2. Valve also will likely remove all 32-bit macOS games from the Steam depot download servers as well if they haven't already, meaning that only 64-bit games will be sold on Steam going forward both in Intel and Silicon binaries, well at least until Rosetta 2 gets removed in a future macOS release which means that only Apple Silicon binaries will be sold on Steam after Rosetta 2 gets removed from macOS later this decade.

**Comment by msikma** [comment]
> Valve will likely drop macOS support from GoldSrc, Source 1 & Source 2 likely later this decade, since they are no longer interested with the platform because of Apple for altering the ecosystem, such as only supporting the Metal API and the Silicon Mac systems, since Apple wants both Vulkan and MoltenVK on macOS to fail NOT succeed.

I think at this point it's best to stick to statements that can be backed up with some sort of direct quote or indication from Valve, rather than just speculating about it. There's no shortage of speculation on this topic at this point, but as far as I know there has never been any official communication on this. It's not that I mind speculation but at least it should be qualified as such.

Besides that, I also want to remind people that older versions of the Source engine have been compiled for native ARM64 just fine, so supporting modern macOS would not be limited to just however long the x86_64 version lasts.

**Comment by Wadmodder** [comment]
> Besides that, I also want to remind people that older versions of the Source engine have been compiled for native ARM64 just fine, so supporting modern macOS would not be limited to just however long the x86_64 version lasts.

GoldSrc & Source 1 use OpenGL on macOS and not Metal API by Apple. The OpenGL implementation on macOS hasn't been updated since version 4.1 released in 2010. In addition, Valve wouldn't want to have the resources to make a OpenGL to MoltenVK translation layer or a MoltenVK to Metal translation layer for either GoldSrc or Source 1, meaning Valve doesn't want to support Apple's Metal API in their engines. Besides, CS2 discontinued support for macOS, meaning that future Valve games will no longer be getting a macOS port in either x86-64 or Apple Silicon ARM binaries.

And besides trying to implement Metal or MoltenVK in GoldSrc or Source 1 on macOS would require rewriting the entire Source branch of TF2, Left 4 Dead, Alien Swarm, Portal 2 and the CSGO Legacy version, and Valve is likely firmly against this, and wouldn't want to dedicate time or resources to do so, much like Valve has been indefinitely delaying the release for the 64-bit exclusive updates for the Steam client for decades. This decade where Windows 11 dumped support for 32-bit processors and the end of support deadline is 2025 for non-LTSC Windows 10 users, and 2032 for the remaining Windows 10 lifespan, means that the 64-bit exclusive updates utilizing 64-bit exclusive Chromium Embedded Framework code will likely be years away of being announced when Valve decides to drop support for 32-bit Windows 10 and earlier users.

So, NO! It doesn't seem that Valve will ever release the 64-bit updates for their other Source 1 games to macOS users, like at all in the future either.

**Comment by Simboiss** [comment]
OpenGL doesn't seem to be a problem right now, assuming Valve still wants to stick to it and nothing else. Apple Silicon is compatible with OpenGL 4.1 and Sonoma still supports it. Then, the next question would be, do Source 1 or GoldSrc absolutely need a version of OpenGL higher than 4.1 to run properly? Is there anything essential to running Source in OpenGL versions 4.2 and later?

An abstraction layer like MoltenGL is likely sufficient to translate from OpenGL to Metal.

**Comment by seanmcgeehan** [comment]
I think we shouldn't pretend that making the mac client would be an impossible task. I haven't worked on this stuff, but from other Major projects i've worked on, I dont think this would take more then 2 weeks considering that this exact work should have already been done for csgo. There should be a template. Any issues that arise should be able to be worked out. 

if about 2 weeks of dev time isn't worth it to valve to support 1-2% of the player base then it is what it is. It, of course, would be profitable to make the mac client.

**Comment by motherofdragons96** [comment]
This is also recent, check out the store page! It doen''t have Mac support anymore!

![image_2024-06-12_161642809](https://github.com/ValveSoftware/Source-1-Games/assets/65467193/225a92e7-8290-45e9-a5d1-f48766b990a4)

**Comment by ethanholt1** [comment]
As of today, [Mac support has been officially removed](https://steamdb.info/app/440/history/?changeid=23945028) from the TF2 page on Steam. This can now be closed.

**Comment by Wadmodder** [comment]
> OpenGL doesn't seem to be a problem right now, assuming Valve still wants to stick to it and nothing else. Apple Silicon is compatible with OpenGL 4.1 and Sonoma still supports it. Then, the next question would be, do Source 1 or GoldSrc absolutely need a version of OpenGL higher than 4.1 to run properly? Is there anything essential to running Source in OpenGL versions 4.2 and later?
> 
> An abstraction layer like MoltenGL is likely sufficient to translate from OpenGL to Metal.

Valve recently also removed the Mac binaries for the other Source Engine titles, including Half-Life 2 (including the episodes, Lost Coast & Deathmatch), Counter-Strike Source, Day of Defeat Source, Half-Life Source (including Deathmatch Source) Portal 1 & 2 and even Left 4 Dead 1 & 2, meaning that the Source SDK is probably going to have the macOS binaries removed as well in the coming months.

And besides, Valve probably didn't want to dedicate time or resources for TF2 on macOS due to the bot crisis either, and they probably disbanded the entire macOS game development team after the release of Half-Life Alyx and the cancellation of SteamVR for macOS.

**Comment by ethanholt1** [comment]
As of today, again, [macOS support has been readded](https://steamdb.info/app/440/history/) to the TF2 app.
However, there is no word whether or not the game functions on macOS.

**Comment by ethanholt1** [comment]
> As of today, again, [macOS support has been readded](https://steamdb.info/app/440/history/) to the TF2 app. However, there is no word whether or not the game functions on macOS.

Just tested on my macOS machine (10.14 Mojave), does not function.

> Valve recently also removed the Mac binaries for the other Source Engine titles, including Half-Life 2 (including the episodes, Lost Coast & Deathmatch), etc...

What? I can still play every single Source game (with the exception of TF2, obviously) just fine. I am updated to the latest available version, and the game binary still exists, and runs fine.

**Comment by cloventt** [comment]
Very disappointing to see macOS support has been abandoned.

**Comment by wokenlex** [comment]
Some problem in what I has payed for this game. I have only Mac. They said before "oh, it's so hard to update to 64bit", but now they did it, and just removed macOS without any public or unpublic notice.

**Comment by Hiiepicsohipeeps2** [comment]
Still no update? Damn

**Comment by hannlol** [comment]
Still nothing? Don't have access to a Windows machine and I would really like to play TF2.

**Comment by m3nt0l** [comment]
The error is still on, when fix please?

**Comment by gaelcoral** [comment]
> The error is still on, when fix please?

I don’t think there will be a fix. They are no longer supporting the Mac version, and even on the TF2 website, they removed the macOS logo, implying that its support has ended

![Image](https://github.com/user-attachments/assets/a8912e49-c876-43cd-80c3-6fa804d1bb53)
